src/lib.rs

```diff
@@ -319,6 +319,19 @@ impl<F: PrimeField, PC: PolynomialCommitment<F, DensePolynomial<F>>, D: Digest>
     ) -> Result<bool, Error<PC::Error>> {
         let verifier_time = start_timer!(|| "Marlin::Verify");
 
+        let public_input = {
+            let domain_x = GeneralEvaluationDomain::<F>::new(public_input.len() + 1).unwrap();
+
+            let mut unpadded_input = public_input.to_vec();
+            let padded_size = domain_x.size();
+
+            if padded_size - 1 > unpadded_input.len() {
+                unpadded_input.resize(padded_size - 1, F::zero());
+            }
```

Comment:
```suggestion
            unpadded_input.resize(core::cmp::max(public_input.len(), domain_x.size() - 1), F::zero());
```

Reply:
That seems to be a better idea. Let me change the implementation accordingly.

Reply:
Nope. Currently we use `format_public_input` and `unformat_public_input` in some places. If we change the logic of `format_public_input`, `unformat_public_input` may have a hard time. It requires more changes.